java/com/google/flatbuffers/FlatBufferBuilder.java

```diff
@@ -368,12 +375,30 @@ public int endVector() {
     * @return The offset in the buffer where the encoded string starts.
     */
     public int createString(String s) {
-        byte[] utf8 = s.getBytes(utf8charset);
-        addByte((byte)0);
-        startVector(1, utf8.length, 1);
-        bb.position(space -= utf8.length);
-        bb.put(utf8, 0, utf8.length);
-        return endVector();
+        return createString(getStringBytes(s));
+    }
+
+    private ByteBuffer getStringBytes(String str) {
```

Comment:
inline this function? especially since calling it twice does not give you two ByteBuffers, it's not that useful for other callers.